frontend/src/Components/SearchBar/SearchBar.jsx

```diff
@@ -1,90 +1,82 @@
-  const [results, setResults] = useState([]);
-  const [showDropdown, setShowDropdown] = useState(false);
-  const latestRequest = useRef(null);
+  const [suggestions, setSuggestions] = useState([]);
 
-  const handleInputChange = async (e) => {
-    const value = e.target.value;
-    setQuery(value);
+  const fetchSuggestions = debounce(async (input) => {
+    if (input.length > 0) {
+      try {
+        const response = await fetch(
+          `http://localhost:3000/api/search?query=${input}&userId=${userId}`
+        );
 
-    if (value.trim() === "") {
-      setResults([]);
-      setShowDropdown(false);
-      return;
-    }
-
-    if (latestRequest.current) {
-      latestRequest.current.abort();
-    }
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
+        }
 
-    const controller = new AbortController();
-    latestRequest.current = controller;
+        const data = await response.json();
 
-    try {
-      const response = await fetch(
-        `http://localhost:3000/api/businesses/search?query=${value}`,
-        { signal: controller.signal }
-      );
-      if (!response.ok) {
-        throw new Error("Failed to fetch search results");
-      }
-      const data = await response.json();
-      setResults(data);
-      setShowDropdown(true);
-    } catch (err) {
-      if (err.name !== "AbortError") {
-        console.error(err);
-        setResults([]);
-        setShowDropdown(false);
+        if (Array.isArray(data)) {
+          setSuggestions(data);
+        } else {
+          setSuggestions([]);
+        }
+      } catch (error) {
+        setSuggestions([]);
       }
+    } else {
+      setSuggestions([]);
     }
+  }, 300);
+
+  const handleChange = (e) => {
+    const input = e.target.value;
+    setQuery(input);
+    fetchSuggestions(input);
   };
 
-  const handleResultClick = (result) => {
-    setQuery(result.name);
-    setShowDropdown(false);
-    onSelect(result);
+  const handleSelect = (suggestion) => {
+    setQuery(suggestion.name);
+    setSuggestions([]);
+    if (onSelect) {
+      onSelect(suggestion);
+    }
   };
 
-  const handleSearchOnMapClick = () => {
-    setShowDropdown(false);
-    onSearchOnMap(query);
+  const clearQuery = () => {
+    setQuery("");
+    setSuggestions([]);
   };
 
   return (
     <div className="search-bar">
       <input
         type="text"
         value={query}
-        onChange={handleInputChange}
-        placeholder="Search for businesses..."
+        onChange={handleChange}
+        placeholder="Search..."
         className="search-input"
       />
-      <button className="search-button">
-        <i className="fa fa-search"></i>
-      </button>
-      {showDropdown && results.length > 0 && (
-        <ul className="search-dropdown">
-          {results.map((result) => (
+      {query && <i className="fa fa-times" onClick={clearQuery}></i>}
+      <i className="fa fa-search"></i>
+      {suggestions.length > 0 && (
```

Comment:
good job checking this! otherwise, we'd end up rendering an empty `<ul>` if suggestions are empty